src/Image.cc

```diff
@@ -420,6 +428,24 @@ Image::loadSurface() {
   if (isJPEG(buf)) return loadJPEG(stream);
 #endif
 
+// svg
+#ifdef HAVE_RSVG
+  // confirm svg using first 1000 chars
+  // if a very long comment precedes the root <svg> tag, isSVG returns false
+  uint8_t head[1000] = {0};
+  fseek(stream, 0 , SEEK_END);
```

Comment:
That's a good point. Perhaps most SVGs do indeed start with xml processing instructions and/or preambles. I don't have a large SVG set to either confirm or discard that. What I have is SVGs for the app I used to work on, but that's meaningless in this context. 

Perhaps it's better to leave it this way then and don't do any premature optimizations at least until someone provides more info. 

Cheers 🍺